feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,769 @@
+		ondatra.CISCO:   regexp.MustCompile("/misc/disk1/.*core.*"),
+		ondatra.NOKIA:   regexp.MustCompile("/var/core/coredump-.*"),
+	}
+)
+
+const (
+	cpuType            = oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_CPU
+	lineCardType       = oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_LINECARD
+	fabricCardType     = oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_FABRIC
+	controllerCardType = oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_CONTROLLER_CARD
+)
+
+// coreFileCheck function is used to check if cores are found on the DUT.
+func coreFileCheck(t *testing.T, dut *ondatra.DUTDevice, gnoiClient raw.GNOI, sysConfigTime uint64, retry bool) {
+	t.Helper()
+	t.Log("Checking for core files on DUT")
+
+	dutVendor := dut.Vendor()
+	// vendorCoreFilePath and vendorCoreProcName should be provided to fetch core file on dut.
+	if _, ok := vendorCoreFilePath[dutVendor]; !ok {
+		t.Fatalf("Please add support for vendor %v in var vendorCoreFilePath ", dutVendor)
+	}
+	if _, ok := vendorCoreFileNamePattern[dutVendor]; !ok {
+		t.Fatalf("Please add support for vendor %v in var vendorCoreFileNamePattern.", dutVendor)
+	}
+
+	in := &fpb.StatRequest{
+		Path: vendorCoreFilePath[dutVendor],
+	}
+	validResponse, err := gnoiClient.File().Stat(context.Background(), in)
+	if err != nil {
+		if retry {
+			t.Logf("Retry GNOI request to check %v for core files on DUT", vendorCoreFilePath[dutVendor])
+			validResponse, err = gnoiClient.File().Stat(context.Background(), in)
+		}
+		if err != nil {
+			t.Fatalf("Unable to stat path %v for core files on DUT, %v", vendorCoreFilePath[dutVendor], err)
+		}
+	}
+
+	// Check cores creation time is greater than test start time.
+	for _, fileStatsInfo := range validResponse.GetStats() {
+		if fileStatsInfo.GetLastModified() > sysConfigTime {
+			coreFileName := fileStatsInfo.GetPath()
+			r := vendorCoreFileNamePattern[dutVendor]
+			if r.MatchString(coreFileName) {
+				t.Logf("INFO: Found core %v on DUT.", coreFileName)
+			}
+		}
+		in = &fpb.StatRequest{
+			Path: fileStatsInfo.GetPath(),
+		}
+		validResponse, err := gnoiClient.File().Stat(context.Background(), in)
+		if err != nil {
+			t.Fatalf("Unable to stat path %v for core files on DUT, %v", vendorCoreFilePath[dutVendor], err)
+		}
+		for _, fileStatsInfo := range validResponse.GetStats() {
+			coreFileName := fileStatsInfo.GetPath()
+			r := vendorCoreFileNamePattern[dutVendor]
+			if r.MatchString(coreFileName) {
+				t.Logf("INFO: Found core %v on DUT.", coreFileName)
+			}
+		}
+	}
+}
+
+func sortedInterfaces(ports []*ondatra.Port) []string {
+	var interfaces []string
+	sort.Slice(ports, func(i, j int) bool {
+		idi, idj := ports[i].ID(), ports[j].ID()
+		li, lj := len(idi), len(idj)
+		if li == lj {
+			return idi < idj
+		}
+		return li < lj // "port2" < "port10"
+	})
+	for _, port := range ports {
+		interfaces = append(interfaces, port.Name())
+	}
+	return interfaces
+}
+
+func contains(list []string, target string) bool {
```

Comment:
I believe some of these helper functions can be replaced/made simple with `slices.Contains`, `slices.Delete`, `slices.Sort` .

Reply:
Thanks for your comments. Import package `Slice`, simplify function `sortedInterfaces`  and remove function `contains`:

```go
	"golang.org/x/exp/slices"
```
```go
func sortedInterfaces(ports []*ondatra.Port) []string {
	var interfaces []string
	for _, port := range ports {
		interfaces = append(interfaces, port.Name())
	}
	slices.Sort(interfaces)
	return interfaces
}
```